docs/implementing-new-trigger.md

```diff
@@ -0,0 +1,71 @@
+# How to add a new event source as Trigger
+
+Kubeless [architecture](/docs/architecture.md) is built on core concepts of Functions, Triggers and Runtime. A _Trigger_ in Kubeless represents association between an event source and functions that need to be invoked on a event in the event source. Kubeless fully leverages the Kubernetes concpets of [custom resource definition](https://kubernetes.io/docs/concepts/api-extension/custom-resources/)(CRD) and [custom controllers](https://kubernetes.io/docs/concepts/api-extension/custom-resources/#custom-controllers). Each trigger is expected to be modelled as Kubernetes CRD. A trigger specific custom resource controller is expected to be written that realizes how deployed functions are invoked when event occurs. Following sections document how one can add a new event source as _Trigger_ into Kubeless.
+
+**Note**: Eventual [vision](https://github.com/kubeless/kubeless/issues/695) of the Kubeless architecture is to let any one develop a Trigger for any event source independent of Kubeless code base. Please see [695](https://github.com/kubeless/kubeless/issues/695) for details. There still work to be done to achieve the end goal. But as a interim solution below guidelines provide a way to add new triggers.
+
+## Model event source as CRD
+
+First step is to create a new CRD for the event source. CRD for the new triggers will be largley similar to the existing ones. For e.g. below is the CRD for Kafka trigger
```

Comment:
`largley` -> `largely` 
`For e.g.` -> `For example` or `e.g.`. (just "e.g." already means "for example")